zerver/webhooks/github/view.py

```diff
@@ -302,16 +317,77 @@ def get_push_commits_body(helper: Helper) -> str:
     )
 
 
+class LazyContext(dict[str, str | int]):
+    """Template rendering context for discussions."""
+
+    def __init__(self, payload: WildValue, include_title: bool) -> None:
+        super().__init__()
+        self.payload = payload
+        self.include_title = include_title
+        self.template_values: dict[str, Callable[[], str | int]] = {
+            "sender": lambda: get_sender_name(self.payload),
+            "author": lambda: self.payload["discussion"]["user"]["login"].tame(check_string),
+            "url": lambda: self.payload["discussion"]["html_url"].tame(check_string),
+            "action": lambda: self.payload["action"].tame(check_string),
+            "optional_title": lambda: f' {self.template_values["title"]()}'
+            if self.include_title
+            else "",
+            "category": lambda: self.payload["discussion"]["category"]["name"].tame(check_string),
+            "title": lambda: self.payload["discussion"]["title"].tame(check_string),
+            "body": lambda: self.payload["discussion"]["body"].tame(check_string),
+            "repository_name": lambda: self.payload["repository"]["name"].tame(check_string),
+            "new_repository_name": lambda: self.payload["changes"]["new_repository"]["name"].tame(
+                check_string
+            ),
+            "discussion_number": lambda: self.payload["discussion"]["number"].tame(check_int),
+            "new_discussion_number": lambda: self.payload["changes"]["new_discussion"][
+                "number"
+            ].tame(check_int),
+            "label": lambda: self.payload["label"]["name"].tame(check_string),
+            "old_category": lambda: self.payload["changes"]["category"]["from"]["name"].tame(
+                check_string
+            ),
+            "locked_reason": lambda: f' as {self.payload["discussion"]["active_lock_reason"].tame(check_string)}'
+            if self.payload["discussion"]["active_lock_reason"]
+            else "",
+            "closed_reason": lambda: self.payload["discussion"]["state_reason"].tame(check_string),
+            "answer_field": lambda: "old_answer"
+            if self.payload["action"].tame(check_string) == "unanswered"
+            else "answer",
+            "answer_url": lambda: self.payload[self.template_values["answer_field"]()][
+                "html_url"
+            ].tame(check_string),
+            "answer_body": lambda: self.payload[self.template_values["answer_field"]()][
+                "body"
+            ].tame(check_string),
+            "comment_id": lambda: self.payload[self.template_values["answer_field"]()]["id"].tame(
+                check_int
+            ),
+        }
+
+    @override
+    def __getitem__(self, key: str) -> str | int:
+        if key in self.template_values:
+            return self.template_values[key]()
+        raise KeyError(f"Key '{key}' not found in LazyContext.")
+
+
 def get_discussion_body(helper: Helper) -> str:
```

Comment:
The other approach would probably involve writing a bunch of `get_discussion_*_body()` functions with the templates inside them, and adding them to the `EVENT_FUNCTION_MAPPER`.
But, I thought this way, we have 2 levels of mapping - 1 at the event level, and 1 at the action level.
And we get to reduce the redundancy of passing the same fields to format() across functions.
We already have a ton of tiny functions in github/view.py, this way we end up with just 1 extra function, a class and a dict constant, and do not add to the pollution.